src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java

```diff
@@ -853,4 +854,50 @@ public void savePaidInFull(Long caseId, PaidInFull paidInFull, String authorisat
             );
         }
     }
+
+    public Claim linkLetterHolder(Long caseId, String letterHolderId) {
+        try {
+            User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+
+            UserDetails userDetails = anonymousCaseWorker.getUserDetails();
+            EventRequestData eventRequestData = eventRequest(LINK_LETTER_HOLDER, userDetails.getId());
+
+            StartEventResponse startEventResponse = startUpdate(
+                anonymousCaseWorker.getAuthorisation(),
+                eventRequestData,
+                caseId,
+                userDetails.isSolicitor() || userDetails.isCaseworker()
+            );
+
+            CCDCase ccdCase = extractCase(startEventResponse.getCaseDetails());
+            Claim claim = caseMapper.from(ccdCase);
+
+            Claim updatedClaim = claim.toBuilder()
```

Comment:
why we are using claim.toBuilder(). The event should have the letterholderId ?

Reply:
it's our existing process of conversion from case to claim